spark/v3.1/spark/src/test/java/org/apache/iceberg/spark/sql/TestDeleteFrom.java

```diff
@@ -41,70 +38,6 @@ public void removeTables() {
     sql("DROP TABLE IF EXISTS %s", tableName);
   }
 
-  @Test
-  public void testDeleteFromUnpartitionedTable() {
-    // This test fails in Spark 3.1. `canDeleteWhere` was added to `SupportsDelete` in Spark 3.1,
-    // but logic to rewrite the query if `canDeleteWhere` returns false was left to be implemented
-    // later.
-    Assume.assumeTrue(Spark3VersionUtil.isSpark30());
```

Comment:
I think Anton added new tests in place of these in 3.2. Can you check whether they should be added here?

Reply:
I added the new tests from v3.2 to the v3.1 `TestDeleteFrom` but the new tests fail. I'll let @aokolnychyi add correct 3.1 versions if applicable in a separate change.